rpc-interface/src/types.rs

```diff
@@ -443,6 +461,33 @@ impl From<nimiq_block::ForkProof> for ForkProof {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum ExecutedTransaction {
+    Ok(Transaction),
+    Err(Transaction),
+}
```

Comment:
This is a major API change again. With this, every transaction will be wrapped into a `Ok: { ...transaction }` object (or `Err: {...}`.

I propose to serde-flatten this enum, so it doesn't wrap. But at the same time, the execution status needs to be conveyed also through the API, so this needs to be "keyed" again, so that the execution status becomes a property on the transaction object, e.g. `executionResult`.